src/database/type-transforms.tsx

```diff
@@ -15,8 +15,16 @@ import {
 import { referenceFacilitiesProp } from ".";
 import { FacilityDocUpdate } from "./types";
 
-const timestampToDate = (timestamp: firebase.firestore.Timestamp): Date => {
-  return timestamp.toDate();
+const timestampToDate = (
+  timestamp: firebase.firestore.Timestamp | null,
+): Date => {
+  if (timestamp) {
+    return timestamp.toDate();
+  } else {
+    // when Firestore writes are pending, server timestamps may be null;
+    // we can substitute the local now in the meantime
```

Comment:
How do we know that this isn't going to be persisted?

Reply:
I will admit that the provenance of these nulls is still mysterious to me ... I haven't really been able figure out why we are getting them. It wasn't such a big deal when they were only happening in `updatedAt` (because we always update that when saving so the previous value wouldn't get persisted anyway) but if it can happen in `createdAt` (it happened to me during one of these duplication operations, that's why I made this change) then it could potentially be persisted. I guess I didn't really think this through all the way? I will try to discern the cause of the null `createdAt` and see if there is a less broad solution

Reply:
Got it. Let me know if you plan on making another change here or if you're ready for another review towards approval.

Reply:
yeah i am still poking around at the null issue, I expect there will be another change here